network/network-address/src/encrypted.rs

```diff
@@ -147,6 +159,83 @@ pub struct EncNetworkAddress {
 #[error("error decrypting network address")]
 pub struct DecryptError;
 
+/////////
```

Comment:
this feels like we should be building a generic AESKey or something in crypto

Reply:
I agree, either `AESKey` or `SymmetricKey`.  Lower down it becomes a little more confusing because you're now using `netaddr::Key` to be specific to the key.  Naming things is hard :/

Reply:
technically this key is not directly used for AES : p

we probably want a more general + consistent approach to storing secrets. for example, not only this key but also the network identity private key(s) should be allocated on the heap, mlock'ed, with guard pages around it https://libsodium.gitbook.io/doc/memory_management

we could probably make a generic `SecretBox<T>` that does this